### PR TITLE
Add Kaufhaus layouts

### PR DESCRIPTION
**Pull request by SSirjes**
Related: https://github.com/Edicy/operations/issues/241
Closes: https://github.com/Voog/design-dusseldorf/issues/85

## Some guides for testing:
- [x]  Check if bgpicker, image drop area and page/article image saving works properly in the following layouts:
  - [x]  **Product list**
  - [x]  **Product**
- [x] Check if **Product list** layout works properly everywhere.
- [x] Check if **Product** layout works properly everywhere.
- [x] Check if **Blog & news** layout didn't broke down.
- [x] Check if **Product list** layout works up to second level in menu.
- [x] Check if **Product list** works for _root item_ (first page).
- [x] Check if **Product** works for _root item_ **(first page)**.
- [x] Check if main menu in desktop didn't broke down.
- [x] Check if main menu in mobile didn't broke down.
- [x] Do a small check for all other layouts to see if something got broken.
- [x] If **first page** is used as **Product list**, there's an option to hide _product_ related pages from main menu. Check if that works as intended.

### CODE REVIEW

**Comment by siimha** [comment]
* [x] Ostukorvi nupul puudub ümbris ja hover class (Vasakul düsseldorf ja paremal anchorage)
http://recordit.co/Fp2rKYoyoO

---

* [x] Variatsiooni valiku nupp liiga väike ja mobiilivaates ümbriseta
* Mobiilivaade
screen shot 2018-01-25 at 12 22 39

* Tavaline vaade
screen shot 2018-01-25 at 12 24 55

---
* [x] Eelvaates variatsiooni valimisel tekib dropdown listi ümber värviline kast:

Mobiilivaates tekib kollane kast:
screen shot 2018-01-25 at 14 12 57 2

Tavalises vaates sinine kast:
screen shot 2018-01-25 at 14 18 07

---
* [x] producti all puudub nupp screen shot 2018-01-25 at 13 30 48 2  ja võimalus lisada alamleht (Anchoragel olemas)

screen shot 2018-01-25 at 13 34 42 2

---
* [x] Product pildi puudumisel ei ole placeholderit (class="item-placeholder")

screen shot 2018-01-25 at 13 55 17 2

---

* ~~Disainitööriistaga ei saa redigeerida toodete nimekirja nagu nt. Longyearbyen-is:
screen shot 2018-01-25 at 14 42 38~~

_**Kommentaar (Sirje)**: see on hea tähelepanek. Võiks eraldi issue raames lahendada._

**Comment by siimha** [comment]
* [ ] Nime ja e-maili lahtrid ei ole kommentaari sisestamisel samal tasemel
screen shot 2018-01-25 at 13 20 14 2

**Comment by SSirjes** [comment]
* [x] kui muuta product list mall product malliks, siis ei kuvata alamlehti.

**Comment by SSirjes** [comment]
* [x] add empty content area for "Product list" layout
* [x] add Add data-search-indexing-allowed="true"
* [x] check submenu logic

**Comment by SSirjes** [comment]
* [x] empty content area should not display
* [x] check common page submenu

**Comment by siimha** [comment]
### Määrates toote malli esileheks:

* ~~Puudub seadistamise võimalus
screen shot 2018-02-13 at 12 53 11~~

Kommentaar(Sirje): võrreldes Sapporo, Satiago ja Longyearbyen kujundusega, siis nendel puudus see võimalus. Kasutaja seiskukohast on hea, kui peamenüüs on koguaeg menüüpunt kuvatud.

* [x]  Headerisse tekib sub-menu
screen shot 2018-02-13 at 16 42 44 2

**Comment by SSirjes** [comment]
* ~~if submenu has no menu items then empty submenu block should not display~~ 
_Separate issue: (https://github.com/Voog/design-dusseldorf/issues/87)_

* [x] check layout_name, type in the manifest file

**Comment by SSirjes** [comment]
* [x] check manifest file, if there is layout_name: page_default and front_page layout_name: page_front

screen shot 2018-02-19 at 13 47 56

**Comment by siimha** [comment]
* [x] variatsiooni valiku väljal ei muutu boarder color sarnaselt buy buttoni värviga (värvi koodid var(--color-dark) / var(--color-light))
screen shot 2018-02-19 at 14 21 47

**Comment by siimha** [comment]
* [x] 600px peal muutub form_field_select laius 100% !important peale, kuid teistel mallidel jääb fikseeritult sama laiusega.
screen shot 2018-02-19 at 14 55 10

**Comment by siimha** [comment]
* ~~Näiteks Anchoragel on custom-btn :hover~~ {
  -webkit-transition: 500ms;
  -o-transition: 500ms;
  transition: 500ms;

http://recordit.co/cgvAHaQtu7

_**Comment (Sirje):** Düsseldorf custom button don't have transition. Design components should look similar_

**Comment by siimha** [comment]
* [x] Crop nupul outline sinine ja lahendus: 
 .bg-crop-btn {  
     &:focus {
        outline: 0;
     }
}
screen shot 2018-02-19 at 16 12 56

**Comment by SSirjes** [comment]
* [x] check buy button variants style on other layouts (common page, front page)

**Comment by siimha** [comment]
* [x] Muuta blog_article.scss faili nimetus (nt. product.scss)
* [x] Content-item.scss faili sisu on juba blog_article.scss failis olemas

**Comment by siimha** [comment]
* ~~Tausta muutmise nupp on product ja product listi vaates madalamal, kui teistel lehtedel.~~

Front/Common/Blog:
screen shot 2018-02-22 at 13 36 26

Product/Product list:
screen shot 2018-02-22 at 13 36 39
 
---

**Comment (Sirje):** kui võrrelda toote detailvaadet ja common page malli, siis nad mõlemad on sarnase ülesehitusega ja bg-pickeri nupp asub samas kohas: http://recordit.co/cR29Wz3a86

Kui võrrelda front page ja product list malle, siis nende ülesehitus on erinev (front page mallil puudub alamlehe lisamise menüü) ja bg picker ei peakski olema samas kohas.

**Comment by siimha** [comment]
* ~~Mina kaotaksin Product listi vaates ülemise alamlehe lisamise nupu.~~
screen shot 2018-02-22 at 13 47 01

_Comment (Sirje): it should stay, so the client can add for product list subpage(s)._

**Comment by siimha** [comment]
* [x] main.scss-is peaks loaderiga seotud koodi stiili muutma. Line 2689-2759 tuleks üle vaadata.
Võiks olla selline: 
screen shot 2018-02-22 at 15 26 13
Ja Grunt ise muudab main.css-is selliseks:
screen shot 2018-02-22 at 15 25 54

Mujal esineb ka sarnast teemat. Grunt lisab automaatselt -webkit-box-flex: 1 ja -ms-flex: 
Hetkel selline:
screen shot 2018-02-22 at 15 37 40

Võiks selline olla:
screen shot 2018-02-22 at 15 38 12

Järgnev näide ka sama teema:
Hetkel selline:
screen shot 2018-02-22 at 15 50 54

Aga võiks olla selline:
screen shot 2018-02-22 at 15 50 48

**Comment by SSirjes** [comment]
* [ ] comment in <script src="{{ javascripts_path }}/application.min.js"></script>
* [ ] manifest fail üle vaadata

**Comment by siimha** [comment]
* [x] Hoveriga kaob osta nupul boarder ära http://recordit.co/UqMkb6NZpj. Mina lahendasin selle järgnevalt (kuigi !important pole just parim lahendus):
screen shot 2018-03-02 at 15 57 51

**Comment by siimha** [comment]
* [x] Lisaksin ka form_field_label klassile var(--color-dark) siis muutuvad sarnaselt ostunupu ja form_field_select värviga (sama teema ka äkki breadcrumbsidega). Anchoragel ka sarnane loogika http://recordit.co/up5fwE2adG

* ~~.bg-crop-btn nupul on sinine outline~~
image

Comment (Sirje): already fixed.

**Comment by siimha** [comment]
* [x] Add CDN for Lazyload (hetkel oled sa Lazyloadi vist otse kopeerinud main.js faili, sest seda ei ole bower componentide all). Realtes https://github.com/Edicy/operations/issues/259 ja lahendus sarnaselt Anchoragele https://github.com/Voog/design-anchorage/pull/285

**Comment by SSirjes** [comment]
* [x] test design editor

**Comment by SSirjes** [comment]
* [x] Product list > product placeholder should be white with dark background

screen shot 2018-03-09 at 09 20 23

* [x] remove white border around product image if product list page has dark background

screen shot 2018-03-09 at 09 56 38

**Comment by SSirjes** [comment]
* [x] List layout submenu should only display under product list page.

**Comment by siimha** [comment]
* [x] form_field_selectile rakendub background-color ja ico-arrowd pole paista.
* [x] form_field_label should have cursor: default style

**Comment by SSirjes** [comment]
* [x] remove unnecessary classes
* [x] replace content-area with formatted

**Comment by SSirjes** [comment]
* [x] disconnect design editor and button text font-size
* [x] add padding-bottom for breadcrumbs menu
* [x] Product list layout breadcrumbs menu padding-bottom fix

**Comment by siimha** [comment]
* [x] Change content-item-box .with-image background background-color opacity to something lower
* [x] Change item-link hover text-decoration: underline to something else

**Comment by SSirjes** [comment]
* [x] comment in <script src="{{ javascripts_path }}/application.min.js"></script>
* [x] manifest fail üle vaadata
* [x] uuendused
* [x] List > product title should be h3
* [x] product > remove h1 extra styles

File: components/submenu.tpl [hunk 1]
```diff
@@ -1,17 +1,76 @@
...
           {% if editmode %}
             {% if item.hidden_children.size > 0 %}<li class="edy-btn">{% menubtn item.hidden_children %}</li>{% endif %}
             <li class="edy-btn">{% menuadd parent="item"%}</li>
           {% endif %}
+        </ul>
+      {% endif %}
+    {% endif %}
+  {% endfor %}
+
+  {% for item in site.all_menuitems_on_level_2 %}
```
Comment (mikkpr) [inline components/submenu.tpl]: Siin ja allpool on hästi palju kordust, millest võiks lahti saada:
```
{% for i in (2..4) %}
  {% assign level_str = 'all_menuitems_on_level_' | append: i %}
  {% for item in site[level_str] %}
    {% if item.selected? and page.level == i | plus: 1  %}
      {% if editmode or item.children? %}
        {% if item.layout_title == 'Product' or item.layout_title == 'Product list' %}
          <ul>
            {% for child in item.visible_children %}
              {% if exclude_products == true and child.layout_title != 'Product' and child.layout_title != "Product list" %}
                <li><a href="{{child.url}}"{% if child.selected? %} class="active"{% endif %}{% unless child.translated? %} class="untranslated fci-editor-menuadd"{% endunless %}>{{child.title}}</a></li>
                {% elsif exclude_products != true %}
                <li><a href="{{child.url}}"{% if child.selected? %} class="active"{% endif %}{% unless child.translated? %} class="untranslated fci-editor-menuadd"{% endunless %}>{{child.title}}</a></li>
              {% endif %}
            {% endfor %}
          </ul>
        {% endif %}
      {% endif %}
    {% endif %}
  {% endfor %}
{% endfor %}
```

Kindlasti peaks läbi testima, aga nii palju kui ise vaatasin, tundus toimivat küll ja on samaväärne kui see praegune variant, kus muutub ainult number.

File: layouts/product.tpl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+      {% include "footer" %}
+    </div>
+  </div>
+  {% include "mobilemenu" %}
+  {% include "javascripts" %}
+  {% include "edicy-tools" items_page: true %}
+
+  {% if editmode %}
+    <script type="text/javascript">
```
Comment (mikkpr) [inline layouts/product.tpl]: Ma arvan, et need custom styled peaks defineerima igas mallis ning kogu see jupp võiks olla kas `javascripts` komponendis või täitsa eraldi komponendis ning layoutides kasutama siis `{% include 'custom-styles' %}` või midagi taolist.

**Review by mikkpr — CHANGES_REQUESTED** [review]
Leidsin paar mõtlemise kohta. Sisuliselt on kõik õige, aga lihtsalt kordust võiks võimalusel vältida.

File: components/submenu.tpl (hunk 1 above)
Comment (siimha) [reply components/submenu.tpl]: Testin lahenduse üle.

Comment (siimha) [reply components/submenu.tpl]: Tegin väikese paranduse ja commitisin ära

File: layouts/product.tpl (hunk 2 above)
Comment (siimha) [reply layouts/product.tpl]: Võtsin scripti layouts failidest ja panin javascript.tpl faili, mis on juba includitud layouts failidesse. Tegin ka commiti.

**Review by mikkpr — APPROVED** [review]

**Pull request merged**